x/ecocredit/server/msg_server.go

```diff
@@ -2,12 +2,14 @@ package server
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/regen-network/regen-ledger/types"
 
 	"github.com/cockroachdb/apd/v2"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/TheBookPeople/iso3166"
```

Comment:
this is GPLv3, I don't think we can use it here unfortunately

Reply:
Oh...that's annoying! Good spot!

Reply:
very unfortunate indeed, there's also https://github.com/pariz/gountries (MIT) but doesn't seem like it's fully complete yet...
thoughts @aaronc? do you believe we should be more simplistic and just use regexp instead (see https://www.iso.org/iso-3166-country-codes.html  "WHAT IS INCLUDED IN ISO 3166?" section) without actually checking if a code really exists?

Reply:
It looks like `gountries` does have enough to make it worthwhile - it's got all the data and we can just construct our own search on the data structure it hands us.

Reply:
Just regex please as I commented elsewhere

Reply:
In the future we can have a dataset managed with x/params maybe but we do not want to force a hard fork